test-data/unit/check-generics.test

```diff
@@ -188,6 +188,71 @@ main:9: error: Incompatible types in assignment (expression has type A[C, B], va
 main:10: error: Incompatible types in assignment (expression has type A[B, B], variable has type A[B, C])
 
 
+[case testClassMethodOfGenericClass]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+U = TypeVar('U')
+class X(Generic[T,U]):
+    def __init__(self, t: T, u: U) -> None:
+        pass
+
+class A(Generic[T]):
+    @classmethod
+    def f(cls, x: T) -> T:
```

Comment:
It's apparently the full type of `A`, namely ``def [T] () -> sm3.A[T`1]``. That `T` is not the same as the `T` in this method, apparently: `a = cls() # type: A[int]` is accepted here. This is maybe a little weird, but not changed by this commit...